qiskit/algorithms/evolvers/variational/__init__.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Variational Quantum Time Evolutions (:mod:`qiskit.algorithms.evolvers.variational`)
+=====================================================
+
+Algorithms for performing Variational Quantum Time Evolution of quantum states and their
+gradients which might be suitable for NISQ devices. VarQTE base class exposes an interface,
+compliant with the Quantum Time Evolution Framework in Qiskit Terra, that is implemented by
+VarQRTE and VarQITE classes for real and imaginary time evolution respectively. The variational
+approach is taken according to a variational principle chosen by a user.
+
+**Examples**
+
+.. code-block::
+
+    # define a Hamiltonian
+    observable = SummedOp(
+        [
+            0.2252 * (I ^ I),
+            0.5716 * (Z ^ Z),
+            0.3435 * (I ^ Z),
+            -0.4347 * (Z ^ I),
+            0.091 * (Y ^ Y),
+            0.091 * (X ^ X),
+        ]
+    ).reduce()
+
+    # define a parametrized initial state to be evolved
+    d = 1
+    ansatz = EfficientSU2(observable.num_qubits, reps=d)
+    parameters = ansatz.ordered_parameters
+
+    # define values of initial parameters
+    init_param_values = np.zeros(len(ansatz.ordered_parameters))
+    for i in range(len(ansatz.ordered_parameters)):
+        init_param_values[i] = np.pi / 2
+    param_dict = dict(zip(parameters, init_param_values))
+
+    # define a variational principle
+    var_principle = ImaginaryMcLachlanVariationalPrinciple()
+
+    # optionally define a backend
+    backend = Aer.get_backend("statevector_simulator")
+
+    # define evolution time
+    time = 1
+
+    # define evolution problem
+    evolution_problem = EvolutionProblem(observable, time, ansatz,
+    hamiltonian_value_dict=param_dict)
+
+    # instantiate the algorithm
+    var_qite = VarQITE(
+        var_principle, backend=backend, error_based_ode=False
+    )
+
+    # run the algorithm/evolve the state
+    evolution_result = var_qite.evolve(evolution_problem)
+
+.. currentmodule:: qiskit.algorithms.evolvers.variational
+
+VarQITE
+--------------------
+
+Algorithm that performs Variational Quantum Imaginary Time Evolution.
+
+.. autosummary::
+   :toctree: ../stubs/
+   :template: autosummary/class_no_inherited_members.rst
+
+   VarQITE
+
+VarQRTE
+--------------------
```

Comment:
The VarQTE base class is a developer artifact and shall not be exposed.

What do you mean by a link to the variational section? I cannot find anything analogous in other init files...

Reply:
VarQTE has the quntum_instance setter in it. Typically we expose classes that the user has visibility to and I think it will also indicate VarQTE as parent.

I meant a link to the modules in `variational` for whatever we need/want to expose to teh user for documentation. In the same was pes_samplers or initial_points now are done in Nature https://github.com/Qiskit/qiskit-nature/blob/762fc0cefa3f18484d32b621b149a99b7803020d/qiskit_nature/algorithms/__init__.py#L97-L103